contracts/SushiYieldSource.sol

```diff
@@ -74,7 +75,7 @@ contract SushiYieldSource is IYieldSource {
 
         uint256 sushiBeforeBalance = sushi.balanceOf(address(this));
 
-        uint requiredShares = ((amount.mul(totalShares) + totalShares.sub(1))).div(barSushiBalance);
+        uint requiredShares = (((amount.mul(totalShares) + totalShares)).div(barSushiBalance)).sub(1);
```

Comment:
what if `(((amount.mul(totalShares) + totalShares)).div(barSushiBalance))` is zero?  That will result in an underflow.  Should there be a special case here?

Reply:
I think the only way you can get 0 in that expression is if the numerator is zero -> the only way the numerator can be zero is if `totalShares` is zero, for which there is a require clause a few lines above.